_posts/software/2020-10-18-autograd.md

```diff
@@ -0,0 +1,139 @@
+
+Control flow:
+
+```python
+>>> def f(x):
+... if x < 0.0:
+...     return f(-x)
+... while x > 10:
+...     x /= 10
+... return np.power(x, 1.5)
+... 
+>>> grad(f)(-10003.0)                                                      
+-0.00015002249831275308
+```
+
+Jacobian:
+
+```python
+>>> from autograd import jacobian
+
+>>> f = lambda x: np.array(
+... [2 * x[0]**2 + 3 * x[1], np.cos(x[0]) + np.sin(x[1])])
+>>> jacobian(f)(np.array([2.0, 3.0]))
+array([[ 8.        ,  3.        ],
+       [-0.90929743, -0.9899925 ]])
+```
+
+Complex-valued function of multiple variables:
+
+```python
+>>> from autograd import holomorphic_grad
+>>> f = lambda x: x[0]*2 + x[1]
+>>> holomorphic_grad(f)(np.array([6.0j, 1.0j]))
+array([2.+0.j, 1.+0.j])
+```
+
+<a id="JAX">
+## But does it scale?!
+
+That's the milion dollar question these days. Luckily for us, the vision of Autograd is being carried on in the high-performance context with [JAX](https://github.com/google/jax), which utilizes XLA (Accelerated Linear Algebra), originally developed for Tensorflow. Although this is still a young library, it promises a lot by combining the best of both worlds.
+
+
+<a id="cite">
+## How do I cite it?
+
+You can cite statsmodels as follows, a pdf of the paper is available <a href="https://indico.ijclab.in2p3.fr/event/2914/contributions/6483/subcontributions/180/attachments/6060/7185/automl-short.pdf" target="_blank">here</a>:
+
+```
+@inproceedings{maclaurin2015autograd,
+  title={Autograd: Effortless gradients in numpy},
+  author={Maclaurin, Dougal and Duvenaud, David and Adams, Ryan P},
+  booktitle={ICML 2015 AutoML Workshop},
+  volume={238},
+  pages={5},
+  year={2015}
+}
+```
+
+<a id="getting-started">
+## How do I get started?
+ 
+ - [Autograd Documentation](https://github.com/HIPS/autograd/blob/master/docs/tutorial.md) has a nice tutorial of implementing logistic regression from start to finish, and lots of other useful information.
+
+<a id="contribute">
+## How do I contribute to the software survey?
+
+<ul>
+  <li><a href="{{ page.annotate_criteria }}" target="_blank">Annotate the software criteria</a></li>
+  <li><a href="{{ page.annotate_taxonomy }}" target="_blank">Annotate the software taxonomy</a></li>
+</ul>
+
+or read more about annotation [here]({{ site.baseurl }}/tutorials/annotate-your-software). You can clone the software repository to do
+bulk annotation, or annotation any repository in the <a href="https://rseng.github.io/software/" target="_blank">software database</a>,
+We want annotation to be fun, straight-forward, and easy, so we will be showcasing one repository to annotate per week.
+If you'd like to request annotation of a particular repository (or addition to the software database)
+please don't hesitate to [open an issue](https://github.com/rseng/software/issues) or even a pull request.
+
+<a id="learn-more">
+## Where can I learn more?
```

Comment:
Learn more shouldn't be changed from the original template - the idea is to learn about the RSEPedia. These links should go up under "Getting Started"

Reply:
Changed it back